File: clients/ios/NabtoShell/Services/ReconnectLogic.swift
import Foundation

struct ReconnectLogic {
    let maxBackoff: TimeInterval = 15
    let maxTotalTime: TimeInterval = 30

    func backoff(attempt: Int) -> TimeInterval {
        let value = pow(2.0, Double(attempt - 1))
        return min(value, maxBackoff)
    }

    func shouldGiveUp(elapsedTime: TimeInterval) -> Bool {
        elapsedTime > maxTotalTime
    }
}

File: clients/android/app/src/main/java/com/nabto/tmuxremote/services/ReconnectLogic.kt
package com.nabto.tmuxremote.services

import kotlin.math.min
import kotlin.math.pow

class ReconnectLogic {
    val maxBackoff: Double = 15.0
    val maxTotalTime: Double = 30.0

    fun backoff(attempt: Int): Double {
        val value = 2.0.pow((attempt - 1).toDouble())
        return min(value, maxBackoff)
    }

    fun shouldGiveUp(elapsedSeconds: Double): Boolean {
        return elapsedSeconds > maxTotalTime
    }
}
